packages/@aws-cdk/aws-events/lib/rule.ts

```diff
@@ -124,19 +127,105 @@ export class Rule extends Resource implements IRule {
    *
    * No-op if target is undefined.
    */
-  public addTarget(target?: IRuleTarget) {
+  public addTarget(target?: IRuleTarget): void {
     if (!target) { return; }
 
     // Simply increment id for each `addTarget` call. This is guaranteed to be unique.
-    const id = `Target${this.targets.length}`;
+    const autoGeneratedId = `Target${this.targets.length}`;
 
-    const targetProps = target.bind(this, id);
+    const targetProps = target.bind(this, autoGeneratedId);
     const inputProps = targetProps.input && targetProps.input.bind(this);
 
     const roleArn = targetProps.role ? targetProps.role.roleArn : undefined;
+    const id = targetProps.id || autoGeneratedId;
+
+    if (targetProps.targetResource) {
+      const targetStack = Stack.of(targetProps.targetResource);
+      const targetAccount = targetStack.account;
+      const targetRegion = targetStack.region;
+
+      const sourceStack = Stack.of(this);
+      const sourceAccount = sourceStack.account;
+
+      if (targetAccount !== sourceAccount) {
+        // cross-account event - strap in, this works differently than regular events!
+        // based on:
+        // https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/CloudWatchEvents-CrossAccountEventDelivery.html
+
+        // for cross-account events, we require concrete accounts
+        if (Token.isUnresolved(targetAccount)) {
+          throw new Error('You need to provide a concrete account for the target stack when using cross-account events');
+        }
+        if (Token.isUnresolved(sourceAccount)) {
+          throw new Error('You need to provide a concrete account for the source stack when using cross-account events');
+        }
+        // and the target region has to be concrete as well
+        if (Token.isUnresolved(targetRegion)) {
+          throw new Error('You need to provide a concrete region for the target stack when using cross-account events');
+        }
+
```

Comment:
As it turns out, there's one more thing we need to check here.  Events cannot be cross-region. Which means we need to check whether the source and target Stacks are in the same region.

Reply:
Done in newest revision.